server/controllers/find/coursesController.test.ts

```diff
@@ -89,13 +86,24 @@ describe('CoursesController', () => {
 
     describe('when some but not all organisations are not found by the organisation service', () => {
       it('renders the course show template with the found organisations', async () => {
-        organisationIds.forEach((organisationId, organisationIdIndex) => {
-          if (organisationIdIndex) {
-            continue
-          }
-          const organisation = organisationFactory.build({ id: organisationId })
-          organisations.push(organisation)
-          when(organisationService.getOrganisation).calledWith(token, organisationId).mockResolvedValue(organisation)
+        const courseOfferings = courseOfferingFactory.buildList(2)
+
+        const courseOfferingsWithUnfindableCourse = [
+          ...courseOfferings,
+          courseOfferingFactory.build({ organisationId: 'NOTFOUND' }),
+        ]
+
+        courseService.getOfferingsByCourse.mockResolvedValue(courseOfferingsWithUnfindableCourse)
+
+        const organisationIds = courseOfferingsWithUnfindableCourse.map(offering => offering.organisationId)
+
+        const organisations = courseOfferingsWithUnfindableCourse.map(offering =>
+          organisationFactory.build({ id: offering.organisationId }),
+        )
```

Comment:
Same with the IDs here - but do we even need to get the IDs? We could just do this and remove the IDs map